packages/core/strapi/lib/services/event-hub.js

```diff
@@ -5,7 +5,28 @@
 
 const EventEmitter = require('events');
 
-class EventHub extends EventEmitter {}
+const { pipeAsync, traverseEntity, sanitize } = require('@strapi/utils');
+
+class EventHub extends EventEmitter {
+  /**
+   * Removes password & private fields from an entity
+   * @param {object | object[]} entity
+   * @param {object} schema
+   * @returns {object}
+   */
+  sanitizeEntity(entity, schema) {
+    if (Array.isArray(entity)) {
+      return Promise.all(entity.map(entry => this.sanitizeEntity(entry, schema)));
+    }
+
+    const sanitizeFunction = pipeAsync(
```

Comment:
Can this be sync or even the tranversEntity is async ?

Reply:
Same comment as later we are leaking quite a few things in there

Reply:
:+1: